src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -23,6 +23,9 @@ public struct Complex : IEquatable<Complex>, IFormattable
         // This is the largest x for which (Hypot(x,x) + x) will not overflow. It is used for branching inside Sqrt.
         private static readonly double s_sqrtRescaleThreshold = double.MaxValue / (Math.Sqrt(2.0) + 1.0);
 
+        // This is the largest x for which 2 x^2 will not overflow. It is used for branching inside Asin and Acos.
+        private static readonly double s_asinOverflowThreshold = Math.Sqrt(double.MaxValue) / 2.0;
```

Comment:
nit: I think this should be a constant. with a comment indicating the operation used to compute the result.

However, that does not follow the existing convention (@mellinoe, do you have a preference here)?